specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,689 @@
+    
+    /// Indicates the kind of context for which the context menu was created
+    /// for the `ICoreWebView2ContextMenuTarget::get_Kind` method.
+    /// This enum will always represent the active element that caused the context menu request. 
+    /// If there is a selection with multiple images, audio and text, for example, the element that 
+    /// the end user right clicks on within this selection will be the option represented by this enum.
+    [v1_enum]
+    typedef enum COREWEBVIEW2_CONTEXT_MENU_TARGET_KIND
+    {
+        /// Indicates that the context menu was created for the page without any additional content.
+        COREWEBVIEW2_CONTEXT_MENU_TARGET_KIND_PAGE,
+
+        /// Indicates that the context menu was created for an image element.
+        COREWEBVIEW2_CONTEXT_MENU_TARGET_KIND_IMAGE,
+
+        /// Indicates that the context menu was created for selected text.
+        COREWEBVIEW2_CONTEXT_MENU_TARGET_KIND_SELECTED_TEXT,
+        
+        /// Indicates that the context menu was created for an audio element.
+        COREWEBVIEW2_CONTEXT_MENU_TARGET_KIND_AUDIO,
+        
+        /// Indicates that the context menu was created for a video element.
+        COREWEBVIEW2_CONTEXT_MENU_TARGET_KIND_VIDEO,
+    } COREWEBVIEW2_CONTEXT_MENU_TARGET_KIND;
+
+    /// Specifies the menu item kind
+    /// for the `ICoreWebView2StagingContextMenuItem::get_Kind` method
+    [v1_enum]
+    typedef enum COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND
+    {
+        /// Specifies a command menu item kind.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_COMMAND,
+
+        /// Specifies a check box menu item kind. `ContextMenuItem` objects of this kind
+        /// will need the `IsChecked` property to determine current state of the check box.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_CHECK_BOX,
+
+        /// Specifies a radio button menu item kind. `ContextMenuItem` objects of this kind
+        /// will need the `IsChecked` property to determine current state of the radio button.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_RADIO,
+
+        /// Specifies a separator menu item kind. `ContextMenuItem` objects of this kind
+        /// are used to signal a visual separator with no functionality.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_SEPARATOR,
+
+        /// Specifies a submenu menu item kind. `ContextMenuItem` objects of this kind will contain
+        /// a `ContextMenuItemCollection` of its children `ContextMenuItem` objects.
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_SUBMENU,
+    } COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND;
+
+    /// Represents a context menu item of a context menu displayed by WebView.
+    [uuid(7aed49e3-a93f-497a-811c-749c6b6b6c65), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuItem : IUnknown
+    {
+        
+        /// Gets the unlocalized name for the `ContextMenuItem`. Use this to distinguish
+        /// between context menu item types.
```

Comment:
Given the confusion about this it would be good to elaborate a bit in this documentation. Something like how this can be used to identify the different built-in context menu items for example the Save As menu item will have a name of '...'. Browser extension context menu items will have a name like ... . Custom context menu items will be named like ...